---
title: "Table 1 Revisions"
author: "Nate Olson"
date: "6/28/2017"
output: html_document
---

Information to add to table 1
## Full taxonomic classification
- complete taxonomic classification of each considered bacterial genus: use taxid classification info? or taxize
```{r}
library(tidyverse)
library(taxize)
genus_names <- c("Bacillus", "Clostridium", "Escherichia",
                                "Francisella", "Listeria", "Pseudomonas",
                                "Salmonella", "Shigella", "Staphylococcus",
                                "Yersinia")
genus_tree <- classification(genus_names, db = "ncbi") %>% class2tree()
```

```{r}
genus_tree$classification %>% arrange(phylum, class, order, family)
```


```{r}
plot(genus_tree, no.margin = TRUE)
tree_df <- fortify(genus_tree$phylo) %>% add_column(tip = .$label) %>% 

      left_join(genus_tree$classification)
library(ggtree)
ggtree(tree_df) + geom_tiplab()
ggsave("genus_tree.pdf")
```   


## Number of species
```{r}
baseline_tbl <- singleOrgMatchResults %>%
      filter(query_genus %in% c("Bacillus", "Clostridium", "Escherichia",
                                "Francisella", "Listeria", "Pseudomonas",
                                "Salmonella", "Shigella", "Staphylococcus",
                                "Yersinia")) %>%
      select(query_genus, Query, query_taxid) %>%
      rename(Taxid = query_taxid) %>% left_join(queryMeta) %>%
      distinct() 
baseline_taxid <- baseline_tbl %>% select(query_genus, Query, Taxid) %>% unique() %>% .$Taxid %>% as.numeric()

baseline_lineage <- taxidClassification[names(taxidClassification) %in% baseline_taxid] %>% 
      bind_rows(.id = "Taxid")
```

```{r}
lineage_df <- baseline_lineage %>% 
      filter(rank %in% c("family", "genus", "species")) %>% 
      select(-id) %>% unique() %>% spread(rank, name)

subspecies_counts <- lineage_df %>% 
      group_by(genus, species) %>% summarise(subspecies_count = n())
species_count <- lineage_df %>% select(genus, species) %>% unique() %>% group_by(genus) %>% summarise(count = n())
species_count
```


## Number of complete sequenced genomes
all but one

## Indication of genome diversity 
- an indication of genome diversity: ANI plots with percent aligned and ANI