---
title: "Table 1 Revisions"
author: "Nate Olson"
date: "6/28/2017"
output: html_document
---

Information to add to table 1
## Full taxonomic classification
- complete taxonomic classification of each considered bacterial genus: use taxid classification info? or taxize
```{r}
library(tidyverse)
library(taxize)
genus_names <- c("Bacillus", "Clostridium", "Escherichia",
                                "Francisella", "Listeria", "Pseudomonas",
                                "Salmonella", "Shigella", "Staphylococcus",
                                "Yersinia")
genus_tree <- classification(genus_names, db = "ncbi") %>% class2tree()
```

```{r}
genus_tree$classification %>% arrange(phylum, class, order, family)
```


```{r}
plot(genus_tree, no.margin = TRUE)
tree_df <- fortify(genus_tree$phylo) %>% add_column(tip = .$label) %>% 

      left_join(genus_tree$classification)
library(ggtree)
ggtree(tree_df) + geom_tiplab()
ggsave("genus_tree.pdf")
```   



- An indication of the genus content and diversity
      - number of distinct species: use species taxid for each entry
      - number of complete sequenced genomes: I think this is the total number of genomes used?
      - an indication of genome diversity: ANI plots with percent aligned and ANI

