---
title: "Table 1 Revisions"
author: "Nate Olson"
date: "6/28/2017"
output: html_document
---

Information to add to table 1
## Full taxonomic classification
- complete taxonomic classification of each considered bacterial genus: use taxid classification info? or taxize
```{r}
library(tidyverse)
library(taxize)
genus_names <- c("Bacillus", "Clostridium", "Escherichia",
                                "Francisella", "Listeria", "Pseudomonas",
                                "Salmonella", "Shigella", "Staphylococcus",
                                "Yersinia")
genus_tree <- classification(genus_names, db = "ncbi") %>% class2tree()
```

```{r}
genus_tree$classification %>% arrange(phylum, class, order, family)
```


```{r}
# plot(genus_tree, no.margin = TRUE)
library(ggtree)
tree_df <- fortify(genus_tree$phylo) %>% add_column(tip = .$label) %>% 
      left_join(genus_tree$classification)

ggtree(tree_df) + geom_tiplab()
ggsave("genus_tree.pdf",width = 7, height = 3.5)
```   


## Number of species
```{r}
baseline_tbl <- singleOrgMatchResults %>%
      filter(query_genus %in% c("Bacillus", "Clostridium", "Escherichia",
                                "Francisella", "Listeria", "Pseudomonas",
                                "Salmonella", "Shigella", "Staphylococcus",
                                "Yersinia")) %>%
      select(query_genus, Query, query_taxid) %>%
      rename(Taxid = query_taxid) %>% left_join(queryMeta) %>%
      distinct() 
baseline_taxid <- baseline_tbl %>% select(query_genus, Query, Taxid) %>% unique() %>% .$Taxid %>% as.numeric()

baseline_lineage <- taxidClassification[names(taxidClassification) %in% baseline_taxid] %>% 
      bind_rows(.id = "Taxid")
```

```{r}
lineage_df <- baseline_lineage %>% 
      filter(rank %in% c("family", "genus", "species")) %>% 
      select(-id) %>% unique() %>% spread(rank, name)

subspecies_counts <- lineage_df %>% 
      group_by(genus, species) %>% summarise(subspecies_count = n())
species_count <- lineage_df %>% select(genus, species) %>% unique() %>% group_by(genus) %>% summarise(count = n())
species_count
```

```{r}
lineage_df %>% select(genus, species) %>% unique() %>% group_by(genus) %>% filter(genus == "Bacillus")
```

```{r}
lineage_df %>% filter(genus == "Bacillus")
```

```{r}
singleOrgRef %>% filter(grepl("Bacillus", file))
```


```{r}
lineage_df %>% select(genus, species) %>% unique() %>% group_by(genus) %>% filter(genus == "Salmonella")
```


```{r}
lineage_df %>% unique() %>% group_by(genus) %>% filter(genus == "Staphylococcus")
```

```{r}
singleOrgRef %>% filter(grepl("argenteus", file))
```


## Number of complete sequenced genomes
all but one

## Indication of genome diversity 
- an indication of genome diversity: ANI plots with percent aligned and ANI

Show pairwise ani for within and between genus

```{r}
dnadiff_tidy <- readRDS("../data/dnaDiffTidy.RDS")
dnadiff_ani <- dnadiff_tidy %>% 
      filter(cat_column == "Align:1-to-1" & metric == "AvgIdentity") %>% 
      select(qry_uid, ref_uid, REF) %>% rename(AvgIdentity = REF) %>% 
      mutate(AvgIdentity = as.numeric(AvgIdentity))
dnadiff_aligned <- dnadiff_tidy  %>% filter(metric == "AlignedBases") %>% 
      mutate(AlignedREF = str_extract(REF, "\\(.*%") %>% str_sub(2,5) %>% as.numeric(),
             AlignedQRY = str_extract(QRY, "\\(.*%") %>% str_sub(2,5) %>% as.numeric()) %>% 
      select(ref_genus, qry_genus, qry_uid, ref_uid, AlignedREF, AlignedQRY)
dnadiff_pair <- dnadiff_aligned %>% left_join(dnadiff_ani)
```


Potential separation based on species - not sure relevance to project
```{r}
dnadiff_pair %>% filter(ref_genus == qry_genus) %>%
      ggplot(aes(x = AlignedREF, y = AvgIdentity, color = ref_genus)) + 
      geom_density2d() + 
      facet_wrap(~ref_genus, nrow = 2) + 
      theme_bw() +
      labs(x = "Percent Aligned", y = "Average Nucleotide Identity", color = "Genus")
```